test/overlapped-checker/main.c

```diff
@@ -0,0 +1,23 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "../../deps/uv/include/uv.h"
+#include "../../deps/uv/src/uv-common.h"
```

Comment:
This is to get the definition of `UV_HANDLE_NON_OVERLAPPED_PIPE`, right?

Depending on internal libuv headers is bad because there's no guarantee they stay usable (or even present.) This could start failing at the next libuv upgrade.

Reply:
I've looked into how libuv gets the overlapped information (the `pNtQueryInformationFile` function) and it seems to be using an internal windows API with manually defined structs in `winapi.h`. I could copy the definitions to avoid depending on libuv internal headers, but duplicating a bunch of doesn't seem much better.

Any thoughts?

Reply:
`NtQueryInformationFile()` is not an internal API. The reason libuv defines it manually is because its definition comes from the DDK (driver development kit) which isn't normally installed. The API itself is always available though.

My preference would be to obtain the overlapped flag from the kernel. You can probably get away with a minimal definition of the relevant structs and enums. You'll need `IO_STATUS_BLOCK` and `FILE_MODE_INFORMATION`; their members are basic types.

`FileModeInformation == 16`, you could just pass that as a literal `/* FileModeInformation */ 16`.

You can steal the actual logic from libuv. :-)

https://github.com/libuv/libuv/blob/549f52c4672a4d8198f096efa513a96b20e2f843/src/win/pipe.c#L297-L310

Reply:
Yesterday I rewrote the test script. Now it is a bit more complex but does check if `FILE_FLAG_OVERLAPPED` is set implicitly (the test program deadlocks if not) without depending on libuv internal headers. Can you review it again and let me know if it is an acceptable alternative to using `NtQueryInformationFile` to check the file flags?